assets/src/components/parts/janus-capi-iframe/ExternalActivity.tsx

```diff
@@ -426,9 +426,22 @@ const ExternalActivity: React.FC<PartComponentProps<CapiIframeModel>> = (props)
                 context: payload.mode,
                 questionId: payload.currentActivityId,
               };
+              const initStateFacts = payload.initStateFacts;
+              let processedInitStateFacts = {};
+              if (scriptEnv) {
+                processedInitStateFacts = Object.keys(initStateFacts).reduce(
+                  (acc: any, key: string) => {
+                    const target = key.split('|')[1];
+                    acc[target] = scriptEnv.Get(key).toJS();
```

Comment:
hmm I was wondering why we were changing it, and now I remember. one problem I have is that I'm not sure that I like how this forces each component to strip out the screen id. I do see *why* that might be necessary, however. it might be possible to do it without doing this though, I'm pretty sure that the reference env that is getting sent to each activity is a child of the defaultGlobalEnv meaning that whatever changes happen to it remain within that context, but it still has read access to the parent context. that being the case we should be able to instead just populate that environment with the localized copies of the variables. also I think the preferred way to get the value would be to use `getValue` from scripting